src/uct/tcp/tcp_ep.c

```diff
@@ -58,3 +60,65 @@ UCS_CLASS_DEFINE(uct_tcp_ep_t, uct_base_ep_t);
 UCS_CLASS_DEFINE_NEW_FUNC(uct_tcp_ep_t, uct_ep_t, uct_iface_t *,
                           const uct_device_addr_t *, const uct_iface_addr_t *);
 UCS_CLASS_DEFINE_DELETE_FUNC(uct_tcp_ep_t, uct_ep_t);
+
+ssize_t uct_tcp_ep_am_bcopy(uct_ep_h uct_ep, uint8_t am_id,
+                            uct_pack_callback_t pack_cb, void *arg)
+{
+    uct_tcp_ep_t *ep = ucs_derived_of(uct_ep, uct_tcp_ep_t);
+    uct_tcp_iface_t *iface = ucs_derived_of(uct_ep->iface, uct_tcp_iface_t);
+    uct_tcp_am_desc_t *desc;
+    uct_tcp_am_hdr_t *hdr;
+    ucs_status_t status;
+    size_t total_length, send_length;
+    ssize_t ret;
+    void *ptr;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
```

Comment:
not so simple to make same macro for put and get, because one is void and the other returns a value. i can make a macro for put only (used 3 times) but not sure it's worth it.